providers/ldap/ldap_client.go

```diff
@@ -323,12 +324,36 @@ func (l *LClient) GetIdentity(distinguishedName string, scope string) (client.Id
 		return nilIdentity, fmt.Errorf("Error %v creating connection", err)
 	}
 	// Bind before query
+	// If service acc bind fails, and auth is on, return identity formed using DN
 	serviceAccountUsername := getUserExternalID(l.Config.ServiceAccountUsername, l.Config.LoginDomain)
 	err = lConn.Bind(serviceAccountUsername, l.Config.ServiceAccountPassword)
 	if err != nil {
-		return nilIdentity, fmt.Errorf("Error %v in ldap bind", err)
-	}
+		defer lConn.Close()
```

Comment:
You defer close here and on line 357 right outside the if block. Can consolidate into one defer before the if block?